### ISSUE TITLE
health ai

### ISSUE THREAD

**Issue by kiru65** [huggingface/huggingface_hub#3337]
import gradio as gr
import torch
from transformers import AutoTokenizer, AutoModelForCausalLM

# Load model and tokenizer
model_name = "ibm-granite/granite-3.2-2b-instruct"
tokenizer = AutoTokenizer.from_pretrained(model_name)

# Ensure proper device handling
device = "cuda" if torch.cuda.is_available() else "cpu"

model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype=torch.float16 if device == "cuda" else torch.float32,
).to(device)

# Ensure pad token exists
if tokenizer.pad_token is None:
    tokenizer.pad_token = tokenizer.eos_token

def generate_response(prompt, max_length=1024):
    inputs = tokenizer(prompt, return_tensors="pt", truncation=True, max_length=512).to(device)

    with torch.no_grad():
        outputs = model.generate(
            **inputs,
            max_length=max_length,
            temperature=0.7,
            do_sample=True,
            pad_token_id=tokenizer.eos_token_id
        )

    response = tokenizer.decode(outputs[0], skip_special_tokens=True)
    # Remove duplicated prompt if present
    if response.startswith(prompt):
        response = response[len(prompt):].strip()
    return response

def disease_prediction(symptoms):
    prompt = (
        f"Based on the following symptoms, provide possible medical conditions and "
        f"general medication suggestions. Always emphasize the importance of consulting "
        f"a doctor for proper diagnosis.\n\nSymptoms: {symptoms}\n\n"
        f"Possible conditions and recommendations:\n\n"
        f"**IMPORTANT: This is for informational purposes only. "
        f"Please consult a healthcare professional for proper diagnosis and treatment.**\n\n"
        f"Analysis:"
    )
    return generate_response(prompt, max_length=1200)

def treatment_plan(condition, age, gender, medical_history):
    prompt = (
        f"Generate personalized treatment suggestions for the following patient information. "
        f"Include home remedies and general medication guidelines.\n\n"
        f"Medical Condition: {condition}\nAge: {age}\nGender: {gender}\nMedical History: {medical_history}\n\n"
        f"Personalized treatment plan including home remedies and medication guidelines:\n\n"
        f"**IMPORTANT: This is for informational purposes only. "
        f"Please consult a healthcare professional for proper treatment.**\n\n"
        f"Treatment Plan:"
    )
    return generate_response(prompt, max_length=1200)

# Build Gradio UI
with gr.Blocks() as app:
    gr.Markdown("# 🩺 Medical AI Assistant")
    gr.Markdown("⚠️ *Disclaimer:* This is for informational purposes only. Always consult healthcare professionals for medical advice.")

    with gr.Tabs():
        with gr.Tab("Disease Prediction"):
            with gr.Row():
                with gr.Column():
                    symptoms_input = gr.Textbox(
                        label="Enter Symptoms",
                        placeholder="e.g., fever, headache, cough, fatigue...",
                        lines=4
                    )
                    predict_btn = gr.Button("Analyze Symptoms")

                with gr.Column():
                    prediction_output = gr.Textbox(label="Possible Conditions & Recommendations", lines=20)

            predict_btn.click(disease_prediction, inputs=symptoms_input, outputs=prediction_output)

        with gr.Tab("Treatment Plans"):
            with gr.Row():
                with gr.Column():
                    condition_input = gr.Textbox(
                        label="Medical Condition",
                        placeholder="e.g., diabetes, hypertension, migraine...",
                        lines=2
                    )
                    age_input = gr.Number(label="Age", value=30)
                    gender_input = gr.Dropdown(
                        choices=["Male", "Female", "Other"],
                        label="Gender",
                        value="Male"
                    )
                    history_input = gr.Textbox(
                        label="Medical History",
                        placeholder="Previous conditions, allergies, medications or None",
                        lines=3
                    )
                    plan_btn = gr.Button("Generate Treatment Plan")

                with gr.Column():
                    plan_output = gr.Textbox(label="Personalized Treatment Plan", lines=20)

            plan_btn.click(
                treatment_plan,
                inputs=[condition_input, age_input, gender_input, history_input],
                outputs=plan_output
            )

# Launch app
app.launch(share=True)

**Comment by Wauplin** [comment]
Hi @kiru65 can you share more context about this thread?
